### DISCUSSION TITLE
Object detection with stitching

### DISCUSSION THREAD

**Discussion by meliioko** [OpenStitching/stitching#193] [Q&A] [answered]
Hello, I am working on a model of object detection, on mutli cameras so i would like to stitch the images of the cameras. The model gives back the coordinate of the box surrounding the object, and i would like to map those cordinates to the panorama given back by the Stitcher but I'm having trouble doing it. Has anyone worked on a similar task ?

**Comment by lukasalexanderweber** [comment, marked as answer]
You have the cooridinates in a single Image and plot them on the final stitched result? The key might be `perspectiveTransform`

See https://stackoverflow.com/questions/76003489/how-to-get-image-coordinates-after-successful-match-and-stitch-using-opencv

**Reply by meliioko** [reply]
Thank you for your fast answer, it seems indeed it is the function I needed.
However, I'm strugling accessing the homography transform after the stitching process was applied, can you explain to me what i could do to acces it ? 
Thank you very much for you help !

**Reply by lukasalexanderweber** [reply]
You can customize the workflow. See https://github.com/OpenStitching/stitching/blob/main/stitching/stitcher.py and https://github.com/OpenStitching/stitching_tutorial/blob/master/docs/Stitching%20Tutorial.md

**Reply by meliioko** [reply]
I have succeeded in retrieving the coordinates in the panorama. Thanks a lot. I wonder if it could be a feature we add to this project, maybe someone else will need it in the future, I guess multi camera tracking is not something niche.

**Reply by lukasalexanderweber** [reply]
I would be glad if you share your solution. We can put the samples [here](https://github.com/OpenStitching/images) and can create a faq content with your code in the discussions or put it in a tutorial. See e.g. the [VideoStitcher](https://github.com/OpenStitching/stitching/discussions/148). While other people might want to stitch videos, I still think it's better when user create their own specific `stitcher` classes for their specific use case.

**Reply by lukasalexanderweber** [reply]
You can also demonstrate it with existing samples. If I understood correctly you had a Rectangle on a single input image:

![inputs](https://github.com/OpenStitching/stitching/assets/32765578/a450d8e1-5478-4c48-b238-fa34bc407de2)

and transformed the coordinates to plot the Rectangle on the final result

![panorama](https://github.com/OpenStitching/stitching/assets/32765578/5581f59f-ec76-479f-8647-2d90580748ad)

**Reply by lukasalexanderweber** [reply]
@meliioko where is your sollution? Don't you want to contribute?
